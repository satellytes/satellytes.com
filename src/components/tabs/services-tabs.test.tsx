import React from 'react';
import ServicesTabs from './services-tabs';
import { render, fireEvent } from '@testing-library/react';

describe('Services Tabs', () => {
  it('should render', () => {
    const servicesTabs = render(<ServicesTabs />);
    expect(servicesTabs).toBeTruthy();
  });

  it('should show UX/UI tab', () => {
    const { getByText } = render(<ServicesTabs />);
    expect(getByText('UX/UI')).toBeTruthy();
  });

  it('should show Full Stack tab', () => {
    const { getByText } = render(<ServicesTabs />);
    expect(getByText('Full Stack')).toBeTruthy();
  });

  it('should show Service Development tab', () => {
    const { getByText } = render(<ServicesTabs />);
    expect(getByText('Service Development')).toBeTruthy();
  });

  it('should show More tab', () => {
    const { getByText } = render(<ServicesTabs />);
    expect(getByText('More')).toBeTruthy();
  });

  it('should focus Service Development tab heading when clicked', () => {
    const { getByRole } = render(<ServicesTabs />);
    fireEvent.click(getByRole('tab', { name: 'Service Development' }));
    expect(getByRole('tab', { name: 'Service Development' })).toHaveFocus();
  });

  it('should activate Service Development tab content, when Service Development tab clicked', () => {
    const { getByRole } = render(<ServicesTabs />);
    fireEvent.click(getByRole('tab', { name: 'Service Development' }));
    expect(
      getByRole('tabpanel', { name: 'Service Development' }),
    ).toBeVisible();
  });
});
