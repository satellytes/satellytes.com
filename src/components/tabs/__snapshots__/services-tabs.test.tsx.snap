// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services Tabs should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-orientation="horizontal"
        data-reach-tabs=""
      >
        <div
          aria-orientation="horizontal"
          class="sc-AxiKw cwqAMm"
          data-reach-tab-list=""
          role="tablist"
        >
          <button
            aria-controls="tabs--20--panel--0"
            aria-selected="true"
            class="sc-AxhCb cFjGRL"
            data-orientation="horizontal"
            data-reach-tab=""
            data-selected=""
            id="tabs--20--tab--0"
            role="tab"
            tabindex="0"
            type="button"
          >
            UX/UI
          </button>
          <button
            aria-controls="tabs--20--panel--1"
            aria-selected="false"
            class="sc-AxhCb cFjGRL"
            data-orientation="horizontal"
            data-reach-tab=""
            id="tabs--20--tab--1"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Full Stack
          </button>
          <button
            aria-controls="tabs--20--panel--2"
            aria-selected="false"
            class="sc-AxhCb cFjGRL"
            data-orientation="horizontal"
            data-reach-tab=""
            id="tabs--20--tab--2"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Service Development
          </button>
          <button
            aria-controls="tabs--20--panel--3"
            aria-selected="false"
            class="sc-AxhCb cFjGRL"
            data-orientation="horizontal"
            data-reach-tab=""
            id="tabs--20--tab--3"
            role="tab"
            tabindex="-1"
            type="button"
          >
            More
          </button>
        </div>
        <div
          class="sc-AxhUy kOIRNg"
          data-reach-tab-panels=""
        >
          <div
            aria-labelledby="tabs--20--tab--0"
            data-reach-tab-panel=""
            id="tabs--20--panel--0"
            role="tabpanel"
            tabindex="0"
          >
            <p>
              Mit rund 20 Jahren Erfahrung in der Konzeption und Umsetzung digitaler Interfaces kennen und können wir das gesamte Spektrum der User Experience (UX). Der kleine, sichtbare Teil davon ist das User Interface (UI), der große unsichtbare (UX) jedoch viel relevanter für die Erreichung ihrer wirtschaftlichen Ziele. Denn nur mit benutzerfreundlichen, leicht verständlichen Journeys über alle Endgeräte, Nutzergruppen und Nutzungskontexte hinweg, erreichen die User und somit Sie ihr Ziel.
            </p>
            <ul
              class="sc-AxjAm butCNa"
            >
              <li
                class="sc-AxirZ cbgdPI"
              >
                Informationsarchitektur (und Navigation)
              </li>
              <li
                class="sc-AxirZ cbgdPI"
              >
                Information/Content Design (Readability, Scanability, Consistency, Density, Sizes)
              </li>
              <li
                class="sc-AxirZ cbgdPI"
              >
                User Interface Design (Atomic Design, Pattern Libraries)
              </li>
              <li
                class="sc-AxirZ cbgdPI"
              >
                Interaction Design (Clarity, Ergonomy, Ease/Joy of Use)
              </li>
              <li
                class="sc-AxirZ cbgdPI"
              >
                Brand Experience
              </li>
              <li
                class="sc-AxirZ cbgdPI"
              >
                Accessibility (WAI)
              </li>
              <li
                class="sc-AxirZ cbgdPI"
              >
                Loading, Responsive-Strategien, Performance
              </li>
            </ul>
          </div>
          <div
            aria-labelledby="tabs--20--tab--1"
            data-reach-tab-panel=""
            hidden=""
            id="tabs--20--panel--1"
            role="tabpanel"
            tabindex="-1"
          >
            <p>
              Angular, React, Vue, Node, Polymer, WebComponent, Redux, Atomic Design... - die Liste der Buzzwords nimmt kein Ende. Ständig tauchen neue Bibliotheken auf, die es sich zum Ziel gesetzt haben das Internet (zumindest teilweise) zu revolutionieren und somit sowohl Betrachtern als auch Entwicklern das Leben zu erleichtern. Viele dieser neuen Technologien verbinden ähnliche grundlegende Konzepte wie Modularität und Wiederverwendbarkeit, womit sich die zur Verfügung gestellten Instrumente der unterschiedlichen Technologien einander durchaus ähneln. Umso wichtiger ist es für uns den Überblick zu behalten und die Spreu vom Weizen zu trennen, um für Ihr Projekt genau diejenige Architektur wählen zu können, die eben jene Instrumente bietet die für Ihr Projekt benötigt werden. Dafür haben wir keine Scheu uns mit dem Modernsten des Modernen auseinanderzusetzen, ohne dabei die Produktionsreife eines Frameworks aus den Augen zu lassen.
            </p>
          </div>
          <div
            aria-labelledby="tabs--20--tab--2"
            data-reach-tab-panel=""
            data-testid="services-development-content"
            hidden=""
            id="tabs--20--panel--2"
            role="tabpanel"
            tabindex="-1"
          >
            <p>
              Oft versuchen Unternehmen ihre analogen Services eins zu eins zu digitalisieren. Oder sie orientieren sich am Wettbewerb und wollen alles was die anderen auch haben – nur um auf Nummer sicher zu gehen. Vielleicht ist das aber gar nicht das was ihre Nutzer wollen? Wir gehen gerne mit Ihnen den Weg ab dem ersten Schritt und erarbeiten Services, die zu Ihren KPIs und Kunden passen.
            </p>
          </div>
          <div
            aria-labelledby="tabs--20--tab--3"
            data-reach-tab-panel=""
            hidden=""
            id="tabs--20--panel--3"
            role="tabpanel"
            tabindex="-1"
          >
            <p>
              And more...
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-orientation="horizontal"
      data-reach-tabs=""
    >
      <div
        aria-orientation="horizontal"
        class="sc-AxiKw cwqAMm"
        data-reach-tab-list=""
        role="tablist"
      >
        <button
          aria-controls="tabs--20--panel--0"
          aria-selected="true"
          class="sc-AxhCb cFjGRL"
          data-orientation="horizontal"
          data-reach-tab=""
          data-selected=""
          id="tabs--20--tab--0"
          role="tab"
          tabindex="0"
          type="button"
        >
          UX/UI
        </button>
        <button
          aria-controls="tabs--20--panel--1"
          aria-selected="false"
          class="sc-AxhCb cFjGRL"
          data-orientation="horizontal"
          data-reach-tab=""
          id="tabs--20--tab--1"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Full Stack
        </button>
        <button
          aria-controls="tabs--20--panel--2"
          aria-selected="false"
          class="sc-AxhCb cFjGRL"
          data-orientation="horizontal"
          data-reach-tab=""
          id="tabs--20--tab--2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Service Development
        </button>
        <button
          aria-controls="tabs--20--panel--3"
          aria-selected="false"
          class="sc-AxhCb cFjGRL"
          data-orientation="horizontal"
          data-reach-tab=""
          id="tabs--20--tab--3"
          role="tab"
          tabindex="-1"
          type="button"
        >
          More
        </button>
      </div>
      <div
        class="sc-AxhUy kOIRNg"
        data-reach-tab-panels=""
      >
        <div
          aria-labelledby="tabs--20--tab--0"
          data-reach-tab-panel=""
          id="tabs--20--panel--0"
          role="tabpanel"
          tabindex="0"
        >
          <p>
            Mit rund 20 Jahren Erfahrung in der Konzeption und Umsetzung digitaler Interfaces kennen und können wir das gesamte Spektrum der User Experience (UX). Der kleine, sichtbare Teil davon ist das User Interface (UI), der große unsichtbare (UX) jedoch viel relevanter für die Erreichung ihrer wirtschaftlichen Ziele. Denn nur mit benutzerfreundlichen, leicht verständlichen Journeys über alle Endgeräte, Nutzergruppen und Nutzungskontexte hinweg, erreichen die User und somit Sie ihr Ziel.
          </p>
          <ul
            class="sc-AxjAm butCNa"
          >
            <li
              class="sc-AxirZ cbgdPI"
            >
              Informationsarchitektur (und Navigation)
            </li>
            <li
              class="sc-AxirZ cbgdPI"
            >
              Information/Content Design (Readability, Scanability, Consistency, Density, Sizes)
            </li>
            <li
              class="sc-AxirZ cbgdPI"
            >
              User Interface Design (Atomic Design, Pattern Libraries)
            </li>
            <li
              class="sc-AxirZ cbgdPI"
            >
              Interaction Design (Clarity, Ergonomy, Ease/Joy of Use)
            </li>
            <li
              class="sc-AxirZ cbgdPI"
            >
              Brand Experience
            </li>
            <li
              class="sc-AxirZ cbgdPI"
            >
              Accessibility (WAI)
            </li>
            <li
              class="sc-AxirZ cbgdPI"
            >
              Loading, Responsive-Strategien, Performance
            </li>
          </ul>
        </div>
        <div
          aria-labelledby="tabs--20--tab--1"
          data-reach-tab-panel=""
          hidden=""
          id="tabs--20--panel--1"
          role="tabpanel"
          tabindex="-1"
        >
          <p>
            Angular, React, Vue, Node, Polymer, WebComponent, Redux, Atomic Design... - die Liste der Buzzwords nimmt kein Ende. Ständig tauchen neue Bibliotheken auf, die es sich zum Ziel gesetzt haben das Internet (zumindest teilweise) zu revolutionieren und somit sowohl Betrachtern als auch Entwicklern das Leben zu erleichtern. Viele dieser neuen Technologien verbinden ähnliche grundlegende Konzepte wie Modularität und Wiederverwendbarkeit, womit sich die zur Verfügung gestellten Instrumente der unterschiedlichen Technologien einander durchaus ähneln. Umso wichtiger ist es für uns den Überblick zu behalten und die Spreu vom Weizen zu trennen, um für Ihr Projekt genau diejenige Architektur wählen zu können, die eben jene Instrumente bietet die für Ihr Projekt benötigt werden. Dafür haben wir keine Scheu uns mit dem Modernsten des Modernen auseinanderzusetzen, ohne dabei die Produktionsreife eines Frameworks aus den Augen zu lassen.
          </p>
        </div>
        <div
          aria-labelledby="tabs--20--tab--2"
          data-reach-tab-panel=""
          data-testid="services-development-content"
          hidden=""
          id="tabs--20--panel--2"
          role="tabpanel"
          tabindex="-1"
        >
          <p>
            Oft versuchen Unternehmen ihre analogen Services eins zu eins zu digitalisieren. Oder sie orientieren sich am Wettbewerb und wollen alles was die anderen auch haben – nur um auf Nummer sicher zu gehen. Vielleicht ist das aber gar nicht das was ihre Nutzer wollen? Wir gehen gerne mit Ihnen den Weg ab dem ersten Schritt und erarbeiten Services, die zu Ihren KPIs und Kunden passen.
          </p>
        </div>
        <div
          aria-labelledby="tabs--20--tab--3"
          data-reach-tab-panel=""
          hidden=""
          id="tabs--20--panel--3"
          role="tabpanel"
          tabindex="-1"
        >
          <p>
            And more...
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
